File: SocialAppFlutter/social_app/lib/Models/last_message.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:social_app/Models/message.dart';
import 'package:social_app/Models/user.dart';

part 'json_generated_code/last_message.g.dart';

@JsonSerializable(explicitToJson: true)
class LastMessage extends Message {
  List<User> userDetails;
  String? conversationDescription;
  List<int>? groupImage;
  bool isGroup;
  LastMessage(
      {required this.userDetails,
      required super.id,
      required super.fromUser,
      required super.content,
      required super.datetime,
      required super.conversationId,
      required super.isImage,
      required super.isVideo,
      required super.fromUserDetails,
      required this.conversationDescription,
      required this.groupImage,
      required this.isGroup,
      required super.usersWhoReacted});

  factory LastMessage.fromJson(Map<String, dynamic> json) =>
      _$LastMessageFromJson(json);

  Map<String, dynamic> toJson() => _$LastMessageToJson(this);
}

File: APIs/ChatService/ChatService/DTOs/LastMessage.cs
using ChatService.Models;
using Newtonsoft.Json;

namespace ChatService.DTOs
{
    public class LastMessage
    {
        [JsonProperty("id")]
        public int Id { get; set; }
        [JsonProperty("content")]
        public string Content { get; set; } = null!;
        [JsonProperty("datetime")]
        public DateTime Datetime { get; set; }
        [JsonProperty("userId")]
        public int UserId { get; set; }
        [JsonProperty("userDetails")]
        public IEnumerable<UserObject?> userDetails { get; set; } = null!;
        [JsonProperty("conversationId")]
        public int ConversationId { get; set; }
        [JsonProperty("fromUser")]
        public int FromUser { get; set; }

        [JsonProperty("fromUserDetails")]
        public UserObject FromUserDetails { get; set; } = null!;
        [JsonProperty("isImage")]
        public bool isImage { get; set; }
        [JsonProperty("isVideo")]
        public bool isVideo { get; set; }
        [JsonProperty("conversationDescription")]
        public string? ConversationDescription { get; set; } = null!;
        [JsonProperty("groupImage")]
        public List<int>? GroupImage { get; set; } = null!;
        [JsonProperty("isGroup")]
        public bool IsGroup { get; set; }
        [JsonProperty("usersWhoReacted")]
        public List<UserObject> UsersWhoReacted { get; set; } = new List<UserObject>();
    }
}
